### PR TITLE
hack: use single output dir

### PR DESCRIPTION
**Pull request by crazy-max**
reduce footprint in our gitignore by using a single output dir for build, cross and coverage. also remove unused `cross-out`. structure will look like this:

* `./bin/build`
* `./bin/coverage`
* `./bin/release`

can be overridden using the `DESTDIR` env var. name of this en var seems to be the more approriate in our case: https://www.gnu.org/prep/standards/html_node/DESTDIR.html.

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,25 +13,23 @@ FROM --platform=$BUILDPLATFORM golang:${GO_VERSION}-alpine AS golatest
...
 FROM gobase AS buildx-build
-ARG LDFLAGS="-w -s"
 ARG TARGETPLATFORM
+RUN xx-go --wrap
 RUN --mount=type=bind,target=. \
   --mount=type=cache,target=/root/.cache \
   --mount=type=cache,target=/go/pkg/mod \
   --mount=type=bind,source=/tmp/.ldflags,target=/tmp/.ldflags,from=buildx-version \
-  set -x; xx-go build -ldflags "$(cat /tmp/.ldflags) ${LDFLAGS}" -o /usr/bin/buildx ./cmd/buildx && \
+  make build CGO_ENABLED=$CGO_ENABLED LDFLAGS="$(cat /tmp/.ldflags)" DESTDIR=/usr/bin && \
```
Comment (thaJeztah) [inline Dockerfile:32]: should the target be last there? so

```
make CGO_ENABLED=$CGO_ENABLED LDFLAGS="$(cat /tmp/.ldflags)" DESTDIR=/usr/bin build
```

Also curious; setting `LDFLAGS` here; does `/tmp/.ldflags` contain the `-X` options?

File: Makefile [hunk 2]
```diff
@@ -1,59 +1,83 @@
...
-	export BUILDX_CMD = $(which buildx)
-else
-	$(error "Buildx is required: https://github.com/docker/buildx#installing")
-endif
+PKG := github.com/docker/buildx
+VERSION ?= $(shell git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
+REVISION ?= $(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
+
+CGO_ENABLED ?= 0
+LDFLAGS ?= -s -w -X ${PKG}/version.Version=${VERSION} -X ${PKG}/version.Revision=${REVISION} -X ${PKG}/version.Package=${PKG}
```
Comment (thaJeztah) [inline Makefile:6]: ... because here, it will skip those if they're not set (i.e., it doesn't _append_ these to the existing ones set, correct?

Comment (thaJeztah) [reply Makefile:6]: ^^ ignore this one; I thought `/tmp/.ldflags` was a "generic" LDFLAGS for the platform, but it includes these

File: Dockerfile [hunk 3]
```diff
@@ -13,25 +13,23 @@ FROM --platform=$BUILDPLATFORM golang:${GO_VERSION}-alpine AS golatest
...
 FROM gobase AS buildx-build
-ARG LDFLAGS="-w -s"
 ARG TARGETPLATFORM
+RUN xx-go --wrap
 RUN --mount=type=bind,target=. \
   --mount=type=cache,target=/root/.cache \
   --mount=type=cache,target=/go/pkg/mod \
   --mount=type=bind,source=/tmp/.ldflags,target=/tmp/.ldflags,from=buildx-version \
-  set -x; xx-go build -ldflags "$(cat /tmp/.ldflags) ${LDFLAGS}" -o /usr/bin/buildx ./cmd/buildx && \
+  make build CGO_ENABLED=$CGO_ENABLED LDFLAGS="$(cat /tmp/.ldflags)" DESTDIR=/usr/bin BINNAME=buildx && \
```
Comment (tonistiigi) [inline Dockerfile:32]: Why this change? How should the user know if they need a specific environment for (some of) the make targets. Seems quite messy to combine these. Same for the `buildx-version`

Comment (crazy-max) [reply Dockerfile]: I changed the implementation so it does not require `make` in the Dockerfile. This commit is to avoid drifting with build opts when buildx is packaged in https://github.com/docker/docker-ce-packaging/blob/5683c6bdcabf000cec58677ebb5f411f5410512a/deb/common/rules#L22. Let me know if you prefer to move it in another PR.

File: hack/build [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+: "${GO_PKG=github.com/docker/buildx}"
+: "${GO_LDFLAGS=-X ${GO_PKG}/version.Version=${VERSION} -X ${GO_PKG}/version.Revision=${REVISION} -X ${GO_PKG}/version.Package=${PACKAGE}}"
+: "${GO_STRIP=}"
+
+if [ -n "$GO_STRIP" ]; then
+  GO_LDFLAGS="$GO_LDFLAGS -s -w"
+fi
+
+set -x
+CGO_ENABLED=0 GO111MODULE=on go build -mod vendor -trimpath -ldflags "${GO_LDFLAGS}" -o "${DESTDIR}/docker-buildx" ./cmd/buildx
```
Comment (tonistiigi) [inline hack/build:19]: where is the `CGO_ENABLED` coming from?

**Comment by crazy-max** [comment]
Removed commits mutualizing local and Dockerfile build opts and hash-files script changes as well. Will do that in a follow-up.

**Review by thaJeztah — APPROVED** [review]
LGTM
